packages/protocol/contracts/utils/RioRewardSplitter.sol

```diff
@@ -0,0 +1,263 @@
+    /// @param token_ The address of the liquid restaking token.
+    function initialize(address initialOwner, address token_) external initializer {
+        __Ownable_init(initialOwner);
+        __UUPSUpgradeable_init();
+        __ReentrancyGuard_init();
+        token = token_;
+    }
+
+    /// @notice Checks if the contract has a balance to distribute.
+    /// @param tokenAddress The address of the token.
+    modifier checkHasUndistributedBalance(address tokenAddress) {
+        if (_getUndistributedBalance(tokenAddress) < 1) revert NO_UNDISTRIBUTED_BALANCE(tokenAddress);
+        _;
+    }
+
+    /// @notice Checks if there are active operators.
+    modifier checkHasActiveOperators() {
+        if (operatorRegistry().activeOperatorCount() < 1) revert NO_ACTIVE_OPERATORS();
+        _;
+    }
+
+    /// @notice Checks if the operator has a balance to withdraw.
+    /// @param operatorId The ID of the operator.
+    /// @param tokenAddress The address of the token.
+    modifier checkOperatorHasTokenBalance(uint8 operatorId, address tokenAddress) {
+        if (_getOperatorBalance(operatorId, tokenAddress) < 1) revert OPERATOR_HAS_NO_TOKENS(operatorId, tokenAddress);
+        _;
+    }
+
+    /// @notice Distributes the currently held undistributed ETH to the operators' withdrawal buckets.
+    function distributeETH() external nonReentrant {
+        _distribute(ETH_ADDRESS);
+    }
+
+    /// @notice Distributes the currently held undistributed ERC20 to the operators' withdrawal buckets.
+    /// @param tokenAddress The address of the token.
+    function distributeERC20(address tokenAddress) external nonReentrant {
+        _distribute(tokenAddress);
+    }
+
+    /// @notice Withdraws the operator's ETH balance from their bucket.
+    /// @param operatorId The ID of the operator.
+    function withdrawETH(uint8 operatorId) external nonReentrant {
+        _withdraw(operatorId, ETH_ADDRESS);
+    }
+
+    /// @notice Withdraws the operator's ERC20 balance from their bucket.
+    /// @param operatorId The ID of the operator.
+    /// @param tokenAddress The address of the token.
+    function withdrawERC20(uint8 operatorId, address tokenAddress) external nonReentrant {
+        _withdraw(operatorId, tokenAddress);
+    }
+
+    /// @notice Distributes the currently held undistributed ETH to the operators' withdrawal buckets and then withdraws it.
+    /// @param operatorId The ID of the operator.
+    function distributeAndWithdrawETH(uint8 operatorId) external nonReentrant {
+        _distribute(ETH_ADDRESS);
+        _withdraw(operatorId, ETH_ADDRESS);
+    }
+
+    /// @notice Distributes the currently held undistributed ERC20 to the operators' withdrawal buckets and then withdraws it.
+    /// @param operatorId The ID of the operator.
+    /// @param tokenAddress The address of the token.
+    function distributeAndWithdrawERC20(uint8 operatorId, address tokenAddress) external nonReentrant {
```

Comment:
I don't believe there's anything detrimental that can be done, but I don't think we should allow the withdrawal of ETH through `withdrawERC20` and `distributeAndWithdrawERC20`.